src/com/owncloud/android/ui/fragment/OCFileListFragment.java

```diff
@@ -381,9 +382,34 @@ public void listDirectory(OCFile directory) {
 
             mAdapter.swapDirectory(directory, storageManager);
             if (mFile == null || !mFile.equals(directory)) {
-                mList.setSelectionFromTop(0, 0);
+                imageView.setSelection(0);
             }
             mFile = directory;
+            
+         // decide image vs. file view
+            double countImages = 0;
+            double countFiles = 0;
+            
+            Vector<OCFile> files = storageManager.getFolderContent(directory);
+            for (OCFile file : files){
+                if (!file.isFolder()){
+                    countFiles++;
+                    
+                    if (file.isImage()){
+                        countImages++;
+                    }
+                }
+            }
+            
+            // > 50% Images --> image view
+            // TODO threshold as constant in Preferences
+            if ((countImages / countFiles) >= 0.5){
```

Comment:
Define the "0.5" value as a constant, better if we have it in a common place for being used in FileListListAdapter class and this one.